core/indexer/process/esdtTransactions.go

```diff
@@ -0,0 +1,73 @@
+package process
+
+import (
+	"math/big"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/parsers"
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/process"
+)
+
+type esdtTransactionProcessor struct {
+	esdtOperations      map[string]struct{}
+	argumentParser      process.CallArgumentsParser
+	internalMarshalizer marshal.Marshalizer
+}
+
+func newEsdtTransactionHandler() *esdtTransactionProcessor {
+	esdtTxProc := &esdtTransactionProcessor{
+		argumentParser: parsers.NewCallArgsParser(),
+	}
+
+	esdtTxProc.initESDTOperations()
+
+	return esdtTxProc
+}
+
+func (etp *esdtTransactionProcessor) initESDTOperations() {
+	etp.esdtOperations = map[string]struct{}{
+		core.BuiltInFunctionESDTTransfer: {},
+		core.BuiltInFunctionESDTBurn:     {},
+		core.BuiltInFunctionESDTFreeze:   {},
+		core.BuiltInFunctionESDTUnFreeze: {},
+		core.BuiltInFunctionESDTWipe:     {},
+		core.BuiltInFunctionESDTPause:    {},
+		core.BuiltInFunctionESDTUnPause:  {},
+	}
+}
+
+func (etp *esdtTransactionProcessor) isESDTTx(tx data.TransactionHandler) bool {
+	txData := tx.GetData()
+	function, _, err := etp.argumentParser.ParseData(string(txData))
```

Comment:
add a check that the receiver is the ESDT address for non-transfer operations
Also.. you can check if the transaction is of type `issue@..` and the receiver is ESDT address [thus, a token issue transaction] and index it as well